library/bignum.c

```diff
@@ -1372,16 +1365,19 @@ int mbedtls_mpi_div_mpi( mbedtls_mpi *Q, mbedtls_mpi *R, const mbedtls_mpi *A,
         }
     }
 
+    T1.s = A->s; /* save sign value in case outputs alias inputs */
+
     if( Q != NULL )
     {
+        T2.s = B->s; /* save sign value in case outputs alias inputs */
         MBEDTLS_MPI_CHK( mbedtls_mpi_copy( Q, &Z ) );
-        Q->s = A->s * B->s;
+        Q->s = T1.s ^ T2.s ? -1 : 1;
```

Comment:
isn't `1 + ( T1.s ^ T2.s )` better?

Reply:
I changed it to `( T1.s ^ T2.s ) | 1`
:rocket: